src/lib.rs

```diff
@@ -1954,14 +1971,62 @@ fn connect(addr: SockAddr, domain: Domain, protocol: Option<Protocol>) -> io::Re
         target_os = "netbsd",
         target_os = "openbsd"
     )))]
-    // If the current platform doesn't support nonblocking at creation, enable it after creation
-    socket.set_nonblocking(true)?;
-    match socket.connect(&addr) {
+    let socket = {
+        #[cfg(not(any(
+            target_os = "macos",
+            target_os = "ios",
+            target_os = "tvos",
+            target_os = "watchos",
+            windows,
+            target_os = "aix",
+            target_os = "haiku"
```

Comment:
ESP-IDF also does not support `CLOEXEC`, so `target_os = "espidf"` needs to be listed here.